test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,145 @@
+      validateStatus: (status) => true,
+    ));
+
+    final dioAdapter = DioAdapter(dio: dio);
+    dio.httpClientAdapter = dioAdapter;
+
+    // Set the expected server date for the mock response
+    final serverDate = DateTime.now().toUtc().add(Duration(milliseconds: 6000));
+    final serverDateIsoString = serverDate.toIso8601String();
+
+    // Create the interceptor
+    final interceptor = RequestsSignatureInterceptor(
+      options,
+      dio,
+      getTime: (request) {
+        return DateTime.now().toUtc().millisecondsSinceEpoch ~/ 1000;
+      },
+      getDateHeader: () {
+        return serverDateIsoString;
+      },
+    );
+
+    // Add the interceptor to Dio
+    dio.interceptors.add(interceptor);
+
+    // Register the mock response
+    dioAdapter.onGet(baseURL, (request) {
+      // Respond with unauthorized or ok based on clock skew and tolerance settings
+      if (options.disableAutoRetryOnClockSkew || toleranceMS < clockskewMS) {
+        // Set the expected date header in the response
+        request.reply(HttpStatus.unauthorized, '{}', headers: {
+          'date': [serverDateIsoString]
+        });
+      } else {
+        request.reply(HttpStatus.ok, {});
+      }
+    });
+
+    // Act
+    final response = await dio.get(baseURL);
+
+    // Assert
+    expect(response.statusCode, HttpStatus.unauthorized);
+  });
+
+  test('Auto-retry disabled when clock skew option is disabled', () async {
+    final dio = Dio();
+    final interceptor = _setUpInterceptor(autoRetry: false);
+    dio.interceptors.add(interceptor);
+
+    final response = await dio.get(baseURL);
+
+    expect(response.statusCode, HttpStatus.ok);
+  });
+
+  test('Auto-retry disabled when tolerance is too low', () async {
+    final dio = Dio();
+    final interceptor = _setUpInterceptor(autoRetry: true);
+    dio.interceptors.add(interceptor);
+
+    final response = await dio.get(baseURL);
+
+    expect(response.statusCode, HttpStatus.ok);
+  });
+
+  test('Auto-retry disabled when no date header', () async {
+    final dio = Dio();
+    final interceptor = _setUpInterceptor(autoRetry: true);
+    dio.interceptors.add(interceptor);
+
+    final response = await dio.get(baseURL);
+
+    expect(response.statusCode, HttpStatus.ok);
+  });
+}
+
+RequestsSignatureInterceptor _setUpInterceptor({bool autoRetry = false}) {
+  final String HEADER_NAME = 'X-Signature';
+  final String CLIENT_ID = 'test_client_id';
+  final String CLIENT_SECRET = 'test_client_secret';
+  final String PATTERN = '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}';
+
+  final options = RequestsSignatureOptions(
+    clientId: CLIENT_ID,
+    clientSecret: CLIENT_SECRET,
+    headerName: HEADER_NAME,
+    signaturePattern: PATTERN,
+    disableAutoRetryOnClockSkew: !autoRetry,
+    clockSkew: Duration(milliseconds: 6000),
+  );
+
+  final dio = Dio(BaseOptions(validateStatus: (status) => true));
+  final dioAdapter = DioAdapter(dio: dio);
+  dio.httpClientAdapter = dioAdapter;
```

Comment:
It seems like you don't configure any behavior for the mock server. I assume it will always return 200 OK. I don't understand how this tests anything?